File: frontend/src/types/finding.ts
export type FindingSeverity = "critical" | "high" | "medium" | "low" | "info";

export type FindingType =
  | "discrepancy"
  | "anomaly"
  | "exposure"
  | "missing"
  | "red_flag";

export interface Finding {
  id: string;
  sessionId: string;
  workspaceId: string;
  type: FindingType;
  severity: FindingSeverity;
  title: string;
  description: string;
  documentRefs: string[];
  fieldRefs: string[];
  confidence: number;
  createdAt: string;
}

File: backend/app/models/finding.py
from dataclasses import dataclass, field


@dataclass
class FindingModel:
    id: str
    session_id: str
    workspace_id: str
    type: str
    severity: str
    title: str
    description: str
    document_refs: list[str] = field(default_factory=list)
    field_refs: list[str] = field(default_factory=list)
    confidence: float = 0.0
    created_at: str = ""
